src/Microsoft.AspNetCore.SignalR.Client.Core/HubConnection.cs

```diff
@@ -814,61 +838,82 @@ private void CheckDisposed()
             }
         }
 
-        private async Task RunClosedEvent(Func<Exception, Task> closed, Exception closeException)
+        private async Task TimerLoop(TimerAwaitable timer)
         {
-            // Dispatch to the thread pool before we invoke the user callback
-            await AwaitableThreadPool.Yield();
+            // get them roughly synced
+            timer.Start();
 
-            try
-            {
-                Log.InvokingClosedEventHandler(_logger);
-                await closed.Invoke(closeException);
-            }
-            catch (Exception ex)
+            _nextActivationSendPing = DateTime.UtcNow + SendPingInterval;
```

Comment:
nit: Use the same `UtcNow` value

Reply:
I wrote a bad comment, the lines intend to initialize the timers rather than sync them. As long as they're within a couple seconds it's okay. Changing the comment.